<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>790</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Banking App</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      <widget class="QWidget" name="loginPage">
       <layout class="QVBoxLayout" name="verticalLayout_Login">
        <item>
         <widget class="QLabel" name="logoLabel">
          <property name="minimumSize">
           <size>
            <width>150</width>
            <height>150</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>150</width>
            <height>150</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">margin-bottom: 15px;</string>
          </property>
          <property name="pixmap">
           <pixmap resource="resources.qrc">:/images/bank_logo.png</pixmap>
          </property>
          <property name="scaledContents">
           <bool>true</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="loginTitle">
          <property name="styleSheet">
           <string notr="true">font-size: 20px; font-weight: bold; color: #82878c; margin-bottom: 20px;</string>
          </property>
          <property name="text">
           <string>Login to Your Account</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="usernameInput">
          <property name="styleSheet">
           <string notr="true">padding: 8px; border: 1px solid #bdc3c7; border-radius: 4px;</string>
          </property>
          <property name="placeholderText">
           <string>Username</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="passwordInput">
          <property name="styleSheet">
           <string notr="true">padding: 8px; border: 1px solid #bdc3c7; border-radius: 4px;</string>
          </property>
          <property name="echoMode">
           <enum>QLineEdit::Password</enum>
          </property>
          <property name="placeholderText">
           <string>Password</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="loginButton">
          <property name="styleSheet">
           <string notr="true">
            QPushButton {
              background-color: #3498db;
              color: white;
              border: none;
              padding: 10px;
              border-radius: 4px;
              font-weight: bold;
            }
            QPushButton:hover {
              background-color: #2980b9;
            }
           </string>
          </property>
          <property name="text">
           <string>Login</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="showRegistrationButton">
          <property name="styleSheet">
           <string notr="true">
            QPushButton {
              background-color: #2ecc71;
              color: white;
              border: none;
              padding: 10px;
              border-radius: 4px;
              font-weight: bold;
            }
            QPushButton:hover {
              background-color: #27ae60;
            }
           </string>
          </property>
          <property name="text">
           <string>Register</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="registrationPage">
       <layout class="QVBoxLayout" name="verticalLayout_Register">
        <item>
         <widget class="QLabel" name="logoLabel">
          <property name="minimumSize">
           <size>
            <width>150</width>
            <height>150</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>150</width>
            <height>150</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">margin-bottom: 15px;</string>
          </property>
          <property name="pixmap">
           <pixmap resource="resources.qrc">:/images/bank_logo.png</pixmap>
          </property>
          <property name="scaledContents">
           <bool>true</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="registerTitle">
          <property name="styleSheet">
           <string notr="true">font-size: 20px; font-weight: bold; color: #82878c; margin-bottom: 20px;</string>
          </property>
          <property name="text">
           <string>Create New Account</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="regUsernameInput">
          <property name="styleSheet">
           <string notr="true">padding: 8px; border: 1px solid #bdc3c7; border-radius: 4px;</string>
          </property>
          <property name="placeholderText">
           <string>Username</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="regPasswordInput">
          <property name="styleSheet">
           <string notr="true">padding: 8px; border: 1px solid #bdc3c7; border-radius: 4px;</string>
          </property>
          <property name="echoMode">
           <enum>QLineEdit::Password</enum>
          </property>
          <property name="placeholderText">
           <string>Password</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="registerButton">
          <property name="styleSheet">
           <string notr="true">
            QPushButton {
              background-color: #2ecc71;
              color: white;
              border: none;
              padding: 10px;
              border-radius: 4px;
              font-weight: bold;
            }
            QPushButton:hover {
              background-color: #27ae60;
            }
           </string>
          </property>
          <property name="text">
           <string>Register</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="showLoginButton">
          <property name="styleSheet">
           <string notr="true">
            QPushButton {
              background-color: #e74c3c;
              color: white;
              border: none;
              padding: 10px;
              border-radius: 4px;
              font-weight: bold;
            }
            QPushButton:hover {
              background-color: #c0392b;
            }
           </string>
          </property>
          <property name="text">
           <string>Back to Login</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="accountPage">
       <layout class="QVBoxLayout" name="verticalLayout_Account">
        <item>
         <widget class="QLabel" name="label">
          <property name="styleSheet">
           <string notr="true">font-size: 18pt; font-weight: bold;</string>
          </property>
          <property name="text">
           <string>Account Overview</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="hboxlayout">
          <item>
           <widget class="QPushButton" name="transferFundsButton">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>Transfer Funds</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="budgetButton">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>Budget Tool</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="viewHistoryButton">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>30</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>30</height>
             </size>
            </property>
            <property name="text">
             <string>View History</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="checkingGroup">
          <property name="title">
           <string>Checking Account (****1234)</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="checkingBalanceText">
             <property name="text">
              <string>Balance:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="checkingBalanceLabel">
             <property name="text">
              <string>$500</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLineEdit" name="checkingAmountInput">
             <property name="placeholderText">
              <string>Amount</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QPushButton" name="checkingDepositBtn">
             <property name="text">
              <string>Deposit</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QPushButton" name="checkingWithdrawBtn">
             <property name="text">
              <string>Withdraw</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="savingsGroup">
          <property name="title">
           <string>Savings Account (****5678)</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="savingsBalanceText">
             <property name="text">
              <string>Balance:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="savingsBalanceLabel">
             <property name="text">
              <string>$1500</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLineEdit" name="savingsAmountInput">
             <property name="placeholderText">
              <string>Amount</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QPushButton" name="savingsDepositBtn">
             <property name="text">
              <string>Deposit</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QPushButton" name="savingsWithdrawBtn">
             <property name="text">
              <string>Withdraw</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="logoutButton">
          <property name="text">
           <string>Log Out</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="transferPage">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item alignment="Qt::AlignHCenter">
         <widget class="QLabel" name="transferLabel">
          <property name="font">
           <font>
            <pointsize>18</pointsize>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Transfer Funds</string>
          </property>
         </widget>
        </item>
        <item alignment="Qt::AlignRight">
         <widget class="QPushButton" name="pushButton">
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>30</height>
           </size>
          </property>
          <property name="text">
           <string>Account Info</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="sourceGroup">
          <property name="title">
           <string/>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="0" column="0">
            <widget class="QLabel" name="sourceLabel">
             <property name="maximumSize">
              <size>
               <width>200</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>Source Account:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="sourceDrop">
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>30</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>100</width>
               <height>30</height>
              </size>
             </property>
             <item>
              <property name="text">
               <string>checking</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>savings</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QLabel" name="sourceBalanceLabel">
             <property name="text">
              <string>Balance:</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="destGroup">
          <property name="title">
           <string/>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="1">
            <widget class="QComboBox" name="destDrop">
             <property name="maximumSize">
              <size>
               <width>100</width>
               <height>30</height>
              </size>
             </property>
             <item>
              <property name="text">
               <string>checking</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>savings</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="destLabel">
             <property name="maximumSize">
              <size>
               <width>200</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>Destination Account:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="destBalanceLabel">
             <property name="text">
              <string>Balance: </string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string/>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="amtLabel">
             <property name="maximumSize">
              <size>
               <width>200</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>Amount ($):</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="amtTextBox">
             <property name="maximumSize">
              <size>
               <width>100</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="placeholderText">
              <string>Enter Amount:</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="transferButton">
          <property name="text">
           <string>Transfer</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="budgetPage">
       <layout class="QVBoxLayout" name="verticalLayout_Budget">
        <item>
         <widget class="QLabel" name="label">
          <property name="font">
           <font>
            <pointsize>18</pointsize>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Budget Calculator</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Monthly After-Tax Income</string>
          </property>
          <layout class="QHBoxLayout">
           <item>
            <widget class="QLineEdit" name="incomeInput">
             <property name="placeholderText">
              <string>Enter your income</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="calculateBudget">
             <property name="text">
              <string>Calculate</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Your 50/30/20 Numbers</string>
          </property>
          <layout class="QFormLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Necessities (50%)</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="necessitiesLabel">
             <property name="text">
              <string>$0.00</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Wants (30%)</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="wantsLabel">
             <property name="text">
              <string>$0.00</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Savings/Debt (20%)</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="savingsLabel">
             <property name="text">
              <string>$0.00</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Custom Budget Categories</string>
          </property>
          <layout class="QVBoxLayout">
           <item>
            <widget class="QListWidget" name="customCategoriesList"/>
           </item>
           <item>
            <layout class="QHBoxLayout">
             <item>
              <widget class="QLineEdit" name="customCategoryInput">
               <property name="placeholderText">
                <string>Category Name</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="customAmountInput">
               <property name="placeholderText">
                <string>Amount</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="addCustomCategory">
               <property name="text">
                <string>Add</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="saveBudget">
          <property name="text">
           <string>Save Budget</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="backToAccountButton">
          <property name="text">
           <string>Back to Account</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="transactionHistoryPage">
       <layout class="QVBoxLayout" name="verticalLayout_TransactionHistory">
        <item>
         <widget class="QLabel" name="transactionHistoryLabel">
          <property name="styleSheet">
           <string notr="true">font-size: 18pt; font-weight: bold;</string>
          </property>
          <property name="text">
           <string>Transaction History</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="transactionTable">
          <property name="rowCount">
           <number>0</number>
          </property>
          <property name="columnCount">
           <number>4</number>
          </property>
          <column>
           <property name="text">
            <string>Source Account</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Destination Account</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Amount</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Timestamp</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="backToAccountBtn">
          <property name="text">
           <string>Back to Account</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
